pkg/build/api/v1/types.go

```diff
@@ -402,6 +408,17 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty"`
 }
 
+// JenkinsPipelineBuildStrategy holds parameters specific to a Jenkins Pipeline build.
+type JenkinsPipelineBuildStrategy struct {
```

Comment:
But you don't "run" a Jenkinsfile, you run a Jenkins Pipeline build. The Jenkinsfile is just the input to the build strategy, which is Jenkins Pipeline. This is the terminology used by Jenkins that users will be familiar with. If you look at the Jenkins UI for example, you don't add a Jenkinsfile build you add a Pipeline.